src/components/footer/config.ts

```diff
@@ -0,0 +1,333 @@
+  },
+  companies: [],
+};
+
+export class FooterConfig {
+  brand: Brand;
+  config: FooterConfigInterface;
+  mappedConfig: null | FooterConfigInstance;
+  isProduction: boolean;
+  useAbsoluteUrls: boolean;
+
+  constructor({
+    config,
+    brand,
+    isProduction = false, // TODO: switch default before merge
+    useAbsoluteUrls = true, // TODO: switch default before merge
+  }: {
+    config: FooterConfigInterface;
+    brand: Brand;
+    isProduction?: boolean;
+    useAbsoluteUrls?: boolean;
+  }) {
+    this.brand = brand;
+    this.config = config;
+    this.isProduction = isProduction;
+    this.useAbsoluteUrls = useAbsoluteUrls;
+
+    this.mappedConfig = null;
+  }
+
+  getMappedConfig(): FooterConfigInstance {
+    if (this.mappedConfig) return this.mappedConfig;
+
+    const mappedConfig = {
+      sections: this.config.sections.map(this.mapSectionConfig.bind(this)),
+      apps: {
+        android: this.mapLinkArray(this.config.apps.android),
+        apple: this.mapLinkArray(this.config.apps.apple),
+      },
+      socialMedia: {
+        facebook: this.mapLinkArray(this.config.socialMedia.facebook),
+        instagram: this.mapLinkArray(this.config.socialMedia.instagram),
+        twitter: this.mapLinkArray(this.config.socialMedia.twitter),
+        youtube: this.mapLinkArray(this.config.socialMedia.youtube),
+      },
+      companies: this.mapLinkArray(this.config.companies),
+    };
+    this.mappedConfig = mappedConfig;
+    return this.mappedConfig;
+  }
+
+  private mapSectionConfig(section: LinkSectionConfig) {
+    const mappedSection = {
+      title: this.mapLinkArray(section.title),
+      items: this.mapLinkArray(section.items),
+    };
+
+    return mappedSection;
+  }
+
+  private mapLinkArray(linkConfigArray: LinkConfig[]): LinkInstance[] {
+    return linkConfigArray.map((linkConfig) => this.mapLink(linkConfig));
+  }
+
+  private mapLink(config: LinkConfig): LinkInstance {
+    return new Link({
+      config,
+      brand: this.brand,
+      isProduction: this.isProduction,
+      useAbsoluteUrls: this.useAbsoluteUrls,
+    });
+  }
+}
+
+// !!CMP Link
+class Link {
+  translationKey: string;
+  link?:
+    | never
+    | {
+        en: string;
+        de: string;
+        fr: string;
+        it: string;
+      };
+  target?: never | '_blank';
+  isVisible: boolean;
+  linkProtocol: string;
+  productionDomains: {
+    as24: string;
+    ms24: string;
+  };
+  preProductionDomains: {
+    as24: string;
+    ms24: string;
+  };
+
+  constructor({
+    config,
+    brand,
+    isProduction,
+    useAbsoluteUrls,
+  }: {
+    config: LinkConfig;
+    brand: Brand;
+    isProduction: boolean;
+    useAbsoluteUrls: boolean;
+  }) {
+    this.translationKey = config.translationKey;
+    this.target = config.target;
+    this.isVisible = Link.determineVisibility({
+      visibilitySettings: config.visibilitySettings,
+      brand,
+    });
+    this.productionDomains = {
+      as24: 'www.autoscout24.ch',
```

Comment:
do you need the `www`?

Reply:
Yes I think otherwise there will be a redirect ever time